.github/workflows/deploy.yml

```diff
@@ -0,0 +1,77 @@
+name: Deploy To Server
+
+on:
+  push:
+    # branches: [ set-up-ci-cd ]
+    branches: [ main ]
+  on:
+  pull_request:
```

Comment:
This can be simplified to:

```
on: pull request
```

`opened, synchronize, reopened` are the defaults events for pull requests (see [docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request)):

![image](https://user-images.githubusercontent.com/696529/227835178-4aa251ff-9803-4459-b5ad-136eb2e44874.png)